src/sagas/featureToggle.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Platform } from 'react-native';
+import { UnleashClient } from 'unleash-proxy-client';
+import { get } from 'lodash';
+
+import {
+  takeEvery,
+  all,
+  call,
+  put,
+  cancelled,
+  select,
+  take,
+  fork,
+} from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import { getUniqueId } from 'react-native-device-info';
+import {
+  setUnleashClient,
+  setFeatureToggles,
+  featureToggleInitialized,
+} from '../actions';
+import {
+  UNLEASH_URL,
+  UNLEASH_CLIENT_KEY,
+  UNLEASH_POLLING_INTERVAL,
+  STAGE,
+} from '../constants';
+
+export function* monitorFeatureFlags() {
+  const unleashClient = new UnleashClient({
+    url: UNLEASH_URL,
+    clientKey: UNLEASH_CLIENT_KEY,
+    refreshInterval: UNLEASH_POLLING_INTERVAL,
+    appName: `wallet-mobile-${Platform.OS}`,
+  });
+
+  const options = {
+    userId: getUniqueId(),
+    properties: {
+      platform: Platform.OS,
+      stage: STAGE,
+    },
+  };
+
+  unleashClient.updateContext(options);
+
+  yield put(setUnleashClient(unleashClient));
+
+  // Listeners should be set before unleashClient.start so we don't miss
+  // updates
+  yield fork(setupUnleashListeners, unleashClient);
+  yield call(() => unleashClient.start());
+
+  const featureToggles = mapFeatureToggles(unleashClient.toggles);
+
+  yield put(setFeatureToggles(featureToggles));
+  yield put(featureToggleInitialized());
+}
+
+export function* setupUnleashListeners(unleashClient) {
+  const channel = eventChannel((emitter) => {
+    const listener = (state) => emitter(state);
+
+    unleashClient.on('update', () => emitter({
+      type: 'FEATURE_TOGGLE_UPDATE',
+    }));
+
+    return () => {
+      unleashClient.removeListener('update', listener);
+    };
+  });
+
+  try {
+    while (true) {
+      const message = yield take(channel);
```

Comment:
This saga will setup an [`eventChannel`](https://redux-saga.js.org/docs/advanced/Channels/#using-the-eventchannel-factory-to-connect-to-external-events) that listens for update events from the unleash client and then it will listen messages emitted from this event channel.

So, in this context, message is what we are emitting using the `emitter`, in this case, an action:

```javascript
{
  type: 'FEATURE_TOGGLE_UPDATE`,
}
```

Which in turn, gets consumed by this `takeEvery` effect:

```javascript
takeEvery('FEATURE_TOGGLE_UPDATE', handleToggleUpdate),
```

And handled by the `handleToggleUpdate` saga

This is just a mechanism to transform regular `EventEmitters` into "redux-action-emitters"

Reply:
Yes

Reply:
You use `message.data` but none of the emmiters have this attribute. Only the error emitter has `payload`. Is that correct? Or should you change to `message.payload` below?

Reply:
Wow, sorry! 

I'm not using `err` anywhere, so I didn't notice that `.data` was undefined

FIxed in 8c09cdef81417539b849fb2018d9e238aef3802b